src/main/java/de/unistuttgart/informatik/fius/icge/workbench/swing/SwingView.java

```diff
@@ -196,7 +205,48 @@ private void initLogPanel(JPanel main) {
         JButton clearLogButton = new JButton("clear log");
         clearLogButton.addActionListener(e -> this.clearLog());
         logPanel.add(BorderLayout.SOUTH, clearLogButton);
-        
-        main.add(BorderLayout.EAST, logPanel);
+
+        JPanel subPanel = new JPanel(new BorderLayout());
+        subPanel.add(logPanel);
+
+        subPanel.add(BorderLayout.PAGE_END, initSpeedSlider());
+
+        main.add(BorderLayout.EAST, subPanel);
+    }
+
+    private JSlider initSpeedSlider() {
+        speedSlider = new JSlider(JSlider.HORIZONTAL, 0, 30, 25);
+
+        speedSlider.setMajorTickSpacing(5);
+        speedSlider.setMinorTickSpacing(1);
+        speedSlider.setSnapToTicks(true);
+        speedSlider.setPaintTicks(true);
+        speedSlider.setPaintLabels(true);
+        speedSlider.setInverted(true);
+        return speedSlider;
     }
+
+    /**
+     * Listener for the speed changing slider and setting the speed for each entity
+     * @param Simulation
+     */
+    class SliderListener implements ChangeListener {
+        private Simulation simulation;
+        public SliderListener (Simulation sim)
+        {
+            this.simulation = sim;
+        }
+        public void stateChanged(ChangeEvent e) {
+            JSlider source = (JSlider)e.getSource();
+            if (!source.getValueIsAdjusting()) {
+                int currentTickCount = source.getValue();
+                if (currentTickCount == 0) // can not be 0, but is available due to nicer layout
+                    currentTickCount = 1;
+                for (Entity entity : this.simulation.getEntities()) {
+                    entity.setDelay(currentTickCount);
```

Comment:
I think the speed slider should not change the number of ticks needed for actions , but rather the number of milliseconds per tick. Changing the milliseconds per tick would not interfere with any other unrelated logic. In order to accomplish that, the `Simulation` class needs to be modified such that the number of milliseconds per tick (which is currently a hardcoded 10) can be set.

(Also: Users might use `setDelay` on `Entity`s overwriting this setting.)